streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/HighAvailabilityTaskAssignor.java

```diff
@@ -124,11 +133,20 @@ private static void assignActiveStatefulTasks(final SortedMap<UUID, ClientState>
             ClientState::assignActive,
             (source, destination) -> true
         );
+
+        if (rackAwareTaskAssignor != null && rackAwareTaskAssignor.isPresent() && rackAwareTaskAssignor.get().canEnableRackAwareAssignor()) {
```

Comment:
Just wondering why we would ever pass in `null`? The rack aware assignor is an internal class, so we can always construct it, and in the end we make a decision about using it or not via `canEnableRackAwareAssignor()` that does the the corresponding config anyway? -- I seems simpler to alway pass an instance of the rack aware assignor and simplify the code here?

Reply:
I think currently it's because there are tests not modified and still passes null. I can remove this in my next PR after I modified all `HAAssignor` tests?